edgexfoundry/device-sdk-go
# feat: Enable service metrics

closes #1225

Add sample `ReadCommandsExecuted` service metric to Device Simple example.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  https://github.com/edgexfoundry/edgex-docs/pull/873

## Testing Instructions
Run non-secure EdgeX stack
Build and run Device Simple  (DEBUG logging) from command line
Verify logs contain the following:
- `Metrics Manager started with a report interval of 30s`
- `Registered ReadCommandsExecuted metric for collection when enabled`
- `Publish 1 metrics to the 'edgex/telemetry/device-simple' base topic`
   - Will repeat every 30 seconds

Build and run ASC from command line (DEBUG logging) using metrics-influxdb profile
Verify folloing message logged
- `Transformed Metric to 'ReadCommandsExecuted,service=device-simple counter-count=2 1664835362873240900\n`
   - Count value will vary


## New Dependency Instructions (If applicable)
`github.com/rcrowley/go-metrics` is new to this service but not EdgeX
